web/thesauruses/python/classes.json

```diff
@@ -0,0 +1,129 @@
+{
+  "meta": {
+    "language": "python",
+    "language_version": "3.10",
+    "language_name": "Python"
+  },
+  "categories": {
+    "Defining Classes": [
+      "normal_class",
+      "abstract_class",
+      "interface",
+      "read_only_class",
+      "static_class",
+      "inner_class",
+      "packages",
+      "class_with_generic_type"
+    ],
+    "Adding Private and Public Members": [
+      "private_variables",
+      "protected_variables",
+      "public_variables",
+      "static_variables",
+      "private_functions",
+      "protected_functions",
+      "public_functions",
+      "static_functions"
+    ],
+    "Extending and Implementing Classes": [
+      "extends_class",
+      "extending_interface",
+      "calling_superclass_functions",
+      "overriding_superclass_functions"
+    ],
+    "Creating Objects and Polymorphism": [
+      "instantiating_object",
+      "instantiating_polymorphic_object"
+    ],
+    "Constructors and Deconstructor": [
+      "implement_constructor",
+      "implement_deconstructor"
+    ]
+  },
+  "classes": {
+    "normal_class": {
+      "code": "class class_name:\n  # class body"
+    },
+    "abstract_class": {
+      "code": "from abc import ABC\nclass class_name(ABC):\n  # class body",
+      "comment": "Note that you will need to import ABC in order to be able to use abstract classes."
+    },
+    "interface": {
+      "code": "import abc\nclass class_name:\n   # Way 1\n   def method1(self):\n      # pass\n   # Way 2\n   @abc.abstractmethod\n   def method2(self):\n      raise NotImplementedError",
+      "comment": "Note that way 1 doesn't force the user to define the methods from the abstract class in the derived class, while way 2 enforces that."
+    },
+    "read_only_class": {
+      "code": "",
+      "not-implemented": true
+    },
```

Comment:
If something's not implemented, you can remove the code line. (it cleans it up a bit.) You have a few of those in this file.

Reply:
Sure.